File: client/java/src/avt/MyScreen.java
package avt;

import javax.microedition.lcdui.Graphics;
import javax.microedition.lcdui.Image;
import main.Canvas;
import main.GameMidlet;

public abstract class MyScreen extends AvMain {
   public static int ITEM_HEIGHT = 20;
   public static Image imgLogo;
   public static Image an;
   public static Image ao;
   public static Image ap;
   public static Image aq;
   public int selected_ = 0;
   public static int nMsg = 0;
   public static int hTab = 20;
   public static int wTab;
   public static int hText;
   public static final int[] colorCity = new int[]{4802889, 3092271};
   public static final int[] colorFarmPath = new int[]{14400144, 12689526};

   public void setSelected(int var1, boolean var2) {
      this.selected_ = var1;
   }

   public void setHidePointer(boolean var1) {
      super.isHide_ = var1;
   }

   public void switchToMe() {
      Canvas.clearKeyPressed();
      Canvas.currentMyScreen = this;
      Canvas.instance.setFullScreenMode(true);
   }

   public void paint(Graphics var1) {
      if (Canvas.menuMain == null && Canvas.currentDialog == null && Canvas.currentFace == null && !ChatTextField.isShow) {
         super.paint(var1);
      } else {
         Canvas.resetTrans(var1);
      }

      if (!Session_ME.gI().isConnected()) {
         Canvas.M.drawString(var1, GameMidlet.APP_VERSION, Canvas.af.x, Canvas.af.y, Canvas.af.anchor);
      } else if (Canvas.currentMyScreen == ServerListScr.me || Canvas.currentMyScreen == MiniMap.me) {
         Canvas.M.drawString(var1, Session_ME.gI().k, Canvas.af.x, Canvas.af.y, Canvas.af.anchor);
      }

      Canvas.paint.paintMSG(var1);
   }

   public void paintMain(Graphics var1) {
   }

   public abstract void update();

   public void keyPress(int var1) {
   }

   public void close() {
   }

   public static void repaint() {
      int var0 = Canvas.h + Canvas.hTab;
      if (an == null) {
         an = Image.createImage(Canvas.w, var0);
      }

      Graphics var1 = an.getGraphics();
      if (Canvas.currentMyScreen == DiamondScr.me_) {
         DiamondScr.me_.paintCaro(var1);
      } else if (Canvas.currentMyScreen == TLBoardScr.instance || Canvas.currentMyScreen == PBoardScr.instance) {
         int var2;
         if (Canvas.w < var0) {
            var2 = var0 / 10;
         } else {
            var2 = Canvas.w / 10;
         }

         for(int var3 = 0; var3 < var2; ++var3) {
            var1.setColor(6629892);
            var1.drawRect(Canvas.hw - var3 * var2 - 1, var0 / 2 - var3 * var2, var3 * var2 << 1, var3 * var2 << 1);
            var1.setColor(13399567);
            var1.drawRect(Canvas.hw - var3 * var2, var0 / 2 - var3 * var2 + 1, var3 * var2 << 1, var3 * var2 << 1);
         }
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/MyScreen.cs
using UnityEngine;

public abstract class MyScreen : AvMain
{
	public static int ITEM_HEIGHT = 20;

	public static Image imgLogo;

	public static Image[] imgChat;

	public int selected;

	public static int nMsg = 0;

	public static int hTab = 20;

	public static int wTab;

	public static int hText;

	public static int colorBar = 0;

	public static Color color;

	public static int colorMiniMap = 0;

	public static int colorPark = 8705740;

	public static int[] colorCity = new int[2] { 4802889, 3092271 };

	public static int[] colorFarmPath = new int[2] { 14400144, 12689526 };

	public virtual void setSelected(int se, bool iss)
	{
		selected = se;
	}

	public void action(int sel)
	{
	}

	public virtual void doMenu()
	{
	}

	public virtual void doSetting()
	{
	}

	public virtual void doMenuTab()
	{
	}

	public virtual void close()
	{
	}

	public static int getHTF()
	{
		return Canvas.h - hTab - 17 * AvMain.hd;
	}

	public virtual void setHidePointer(bool iss)
	{
		isHide = iss;
	}

	public virtual void switchToMe()
	{
		Canvas.currentMyScreen = this;
	}

	public virtual void initZoom()
	{
	}

	public override void paint(MyGraphics g)
	{
		if (Canvas.menuMain == null)
		{
			if (Canvas.currentDialog == null && Canvas.currentFace == null && !ChatTextField.isShow)
			{
				base.paint(g);
			}
			else
			{
				Canvas.resetTransNotZoom(g);
			}
			if (!Session_ME.gI().isConnected())
			{
				Canvas.arialFont.drawString(g, "2.5.8", Canvas.posByteCOunt.x, Canvas.posByteCOunt.y, Canvas.posByteCOunt.anchor);
			}
			else if (Canvas.currentMyScreen == ServerListScr.me)
			{
				Canvas.arialFont.drawString(g, Session_ME.strRecvByteCount + string.Empty, Canvas.posByteCOunt.x, Canvas.posByteCOunt.y, Canvas.posByteCOunt.anchor);
			}
		}
	}

	public virtual void paintMain(MyGraphics g)
	{
	}

	public abstract void update();

	public override void keyPress(int keyCode)
	{
	}

	public override void commandTab(int index)
	{
	}

	public override void commandAction(int index)
	{
	}

	public virtual void doChat(string text)
	{
	}

	public void repaint()
	{
	}
}
